src/ostree/ot-admin-builtin-status.c

```diff
@@ -43,31 +43,64 @@ static GOptionEntry options[]
         { NULL } };
 
 static gboolean
-deployment_is_prepared_for_soft_reboot (OstreeDeployment *deployment)
+deployment_is_prepared_for_soft_reboot (OstreeSysroot *sysroot, OstreeDeployment *deployment)
 {
-  // Check if /run/nextroot exists (indicating a soft-reboot is prepared)
+  // Check if /run/ostree/nextroot-booted exists (indicating a soft-reboot is prepared)
+  if (!g_file_test ("/run/ostree/nextroot-booted", G_FILE_TEST_EXISTS))
+    return FALSE;
+
+  // Also verify that /run/nextroot actually contains a deployment
   if (!g_file_test ("/run/nextroot", G_FILE_TEST_IS_DIR))
     return FALSE;
 
-  // Get the deployment info we're looking for
-  const char *deployment_csum = ostree_deployment_get_csum (deployment);
-  int deployment_serial = ostree_deployment_get_deployserial (deployment);
-  const char *deployment_osname = ostree_deployment_get_osname (deployment);
+  // Check if /run/nextroot has actual content (not just an empty directory)
+  if (!g_file_test ("/run/nextroot/sysroot", G_FILE_TEST_IS_DIR))
+    return FALSE;
 
-  // Try to find this deployment under /run/nextroot/sysroot/ostree/deploy
-  g_autofree char *nextroot_deploy_path
-      = g_strdup_printf ("/run/nextroot/sysroot/ostree/deploy/%s/deploy", deployment_osname);
+  // Load the metadata from the nextroot-booted file
+  gchar *metadata_contents = NULL;
+  gsize metadata_length = 0;
+  g_autoptr(GError) local_error = NULL;
+  if (!g_file_get_contents ("/run/ostree/nextroot-booted", &metadata_contents, &metadata_length, &local_error))
```

Comment:
- We don't want to re-parse this file every time we look at a deployment but instead do it in...probably `ostree_sysroot_load_if_changed` (and if we do that we need to be careful to also bump the sysroot mtime when we prep a soft reboot!)
- It's most elegant here to use `ot_openat_ignore_enoent` (many similar uses in the codebase)